src/zope/app/dependable/__init__.py

```diff
@@ -85,12 +85,15 @@ def _make_relative(self, path):
             path = canonicalPath(path)
             if path.startswith(self.pp):
                 path = path[self.pplen:]
-                while path.startswith("/"): # pragma: no cover
-                    # We should not be able to get here. canonicalPath
-                    # doesn't allow trailing / in a path segment, and we
-                    # already cut off the whole length of the parent, which
-                    # we guaranteed to begin and end with a /
-                    path = path[1:]
+                # Now, the path should not actually begin with a /.
+                # canonicalPath doesn't allow trailing / in a path
+                # segment, and we already cut off the whole length of
+                # the parent, which we guaranteed to begin and end
+                # with a /. But it's possible that older dependencies
+                # than we test with could produce this scenario, so we
+                # leave it for BWC.
```

Comment:
> BWC = backwards compatibility? First time I see this acronym.

Indeed. I guess that's what I "grew up" using? Maybe it's more common in Java, or maybe it was just the company culture. In the code I have handy, I find hits for it in :
- zope.proxy
- zope.session
- zc.intid
- relstorage
- gevent 

But to be fair, some of those *may* be me.

> We used to use BBB in Zope-land, which doesn't stand for anything

That's really good to know. I've wracked my brain trying, and failing, to figure out what that meant.